semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/Scalacp.scala

```diff
@@ -159,17 +159,9 @@ object Scalacp {
   }
 
   private def sname(sym: Symbol): String = {
-    def loop(name: String): String = {
-      val i = name.lastIndexOf("$$")
-      if (i > 0) loop(name.substring(i + 2))
-      else if (name.endsWith(" ")) loop(name.substring(0, name.length - 1))
-      else if (name == "<root>") n.RootPackage
-      else if (name == "<empty>") n.EmptyPackage
-      else if (name == "<init>") n.Constructor
-      else if (name == "<refinement>") "$anon"
-      else NameTransformer.decode(name)
-    }
-    loop(sym.name)
+    val ssym = ssymbol(sym)
```

Comment:
⚠️ Warning: handwavy perf comment

This seems like a heavy hammer for a small problem. I see you moved `loop` to another place, I'm not sure exactly what the consequences for this refactoring are but it seems to be doing much more work now for the same functionality with a special case only for package object 🤔